Lib/glyphsLib/builder/custom_params.py

```diff
@@ -251,21 +248,23 @@ def to_ufo(self, builder, glyphs, ufo):
         self._write_to_ufo(glyphs, ufo, ufo_value)
 
     def _read_from_glyphs(self, glyphs):
-        # Is it now a property?
-        if self.glyphs3_property and glyphs.has_properties():
-            return glyphs.get_property(self.glyphs3_property)
-        # Try both the prefixed (long) name and the short name
+        value = None
+        # Try to read from the properties first.
+        value = glyphs.get_property(self.glyphs3_property)
+        if value is not None:
+            return value
+        # Try to read from the custom parameters next, using both the short
+        # name, which has precedence, and the prefixed (long) name.
         if self.glyphs_multivalued:
             getter = glyphs.get_custom_values
         else:
             getter = glyphs.get_custom_value
-        # The value registered using the small name has precedence
-        small_name_value = getter(self.glyphs_name)
-        if small_name_value is not None:
-            return small_name_value
+        value = getter(self.glyphs_name)
+        if value is not None:
+            return value
         if self.glyphs_long_name is not None:
-            return getter(self.glyphs_long_name)
-        return None
+            value = getter(self.glyphs_long_name)
+        return value
```

Comment:
I think what George is saying is that when reading a Glyphs 2 file, we should populate the properties from the corresponding custom parameters, and this code can be simplified to always check for the properties.

Reply:
I see. Yes, this is a better interpretation. Thank you for clarifying 🙂

So the logic should be that, if there is a known property, it should be the only place checked in this code, and that, when parsing, all custom parameters with known properties should go to `properties`. Then when writing files in format 2, one would have to put some properties back to custom parameters instead of ignoring them all, as it is currently.

Reply:
I believe so. But it doesn’t have to happen in the PR, I think this change is fine as it is since it fixes an existing bug.

Reply:
Yes, it is more of a refactoring task, which can be tackled separately.

Reply:
Yes. That is correct. 

That simplifies the object API and keeps the conversation code in two place: the reading and the writing function for each object.